staging/src/k8s.io/kubectl/pkg/describe/versioned/describe.go

```diff
@@ -3913,6 +3914,61 @@ func describeStorageClass(sc *storagev1.StorageClass, events *corev1.EventList)
 	})
 }
 
+type CSINodeDescriber struct {
+	clientset.Interface
+}
+
+func (c *CSINodeDescriber) Describe(namespace, name string, describerSettings describe.DescriberSettings) (string, error) {
+	csi, err := c.StorageV1().CSINodes().Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	var events *corev1.EventList
+	if describerSettings.ShowEvents {
+		events, _ = c.CoreV1().Events(namespace).Search(scheme.Scheme, csi)
+	}
+
+	return describeCSINode(csi, events)
+}
+
+func describeCSINode(csi *storagev1.CSINode, events *corev1.EventList) (output string, err error) {
+	metadata := csi.GetObjectMeta()
+	return tabbedString(func(out io.Writer) error {
+		w := NewPrefixWriter(out)
+		w.Write(LEVEL_0, "Name:\t%s\n", csi.GetName())
+		printLabelsMultiline(w, "Labels", csi.GetLabels())
+		printAnnotationsMultiline(w, "Annotations", csi.GetAnnotations())
+		w.Write(LEVEL_0, "Metadata:\n")
+		w.Write(LEVEL_1, "Creation Timestamp:\t%s\n", csi.CreationTimestamp.Time.Format(time.RFC1123Z))
+		for _, owner := range metadata.GetOwnerReferences() {
+			w.Write(LEVEL_1, "Owner References:\n")
+			w.Write(LEVEL_2, "API Version:\t%s\n", owner.APIVersion)
+			w.Write(LEVEL_2, "Kind:\t%s\n", owner.Kind)
+			w.Write(LEVEL_2, "Name:\t%s\n", owner.Name)
+			w.Write(LEVEL_2, "UID:\t%s\n", owner.UID)
+		}
+		w.Write(LEVEL_1, "Resource Version:\t%s\n", metadata.GetResourceVersion())
+		w.Write(LEVEL_1, "Self Link:\t%s\n", metadata.GetSelfLink())
+		w.Write(LEVEL_1, "UID:\t%s\n", metadata.GetUID())
+		w.Write(LEVEL_0, "Spec:\n")
+		if csi.Spec.Drivers != nil {
+			w.Write(LEVEL_1, "Drivers:\n")
+			for _, driver := range csi.Spec.Drivers {
+				w.Write(LEVEL_2, "Allocatables:\n")
+				w.Write(LEVEL_3, "Count:\t%d\n", *driver.Allocatable.Count)
+				w.Write(LEVEL_2, "Name:\t%s\n", driver.Name)
```

Comment:
I'd put name to the top, without `Name:` and indent it, because it's the most important field and separates items in array:

```
Drivers:
    ebs.csi.aws.com
        Node ID: i-367653765
        Topology Keys: ...
    rbd.csi.ceph.com:
        Node ID: node1.foo.bar.org
```

This is similar to e.g. `kubectl describe pod`:

```
Volumes:
  kubecerts:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/kubernetes/static-pod-resources/etcd-member
    HostPathType:  
  default-token-wcs5x:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-wcs5x
    Optional:    false
```
```

Reply:
I've updated this section to use the suggested layout. Testing with a simple CSINode I see the following now:

```
Spec:
  Drivers:
    csi-mock-csi-mock-volumes-1537:
      Allocatables:
        Count:        2
      Node ID:        csi-mock-csi-mock-volumes-1537
      Topology Keys:  [1 2]
    csi-mock-csi-mock-volumes-5691:
      Allocatables:
        Count:        2
      Node ID:        csi-mock-csi-mock-volumes-5691
      Topology Keys:  []
    csi-mock-csi-mock-volumes-6950:
      Allocatables:
        Count:        2
      Node ID:        csi-mock-csi-mock-volumes-6950
      Topology Keys:  []
```